source/code/source/playfab/PlayFabEventPipeline.cpp

```diff
@@ -184,10 +238,17 @@ namespace PlayFab
                 {
                     // do not take new events from buffer if batches currently in flight are at the maximum allowed number
                     // and are not sent out (or received an error) yet
-                    std::this_thread::sleep_for(std::chrono::milliseconds(this->settings->readBufferWaitTime)); // give some time for batches in flight to deflate
-                    continue;
+                    return false;
                 }
 
+                // Don't try taking a request until we get an entity token. We'd rather not lose events
+                // that were generated before an entity token was created (i.e. before a user was created).
+                if (PlayFabSettings::staticPlayer->entityToken.empty() &&
```

Comment:
I'm torn on this if-block in your change.
in general, I think it's good since..
1. for playfab sdk consumers it enables users of the event pipeline to queue events BEFORE a user logs in
2. it's one less things that's specific to our private fork of the SDK

but I'd be remiss if I didn't speak up and say it's actually not a part of the change to the event pipeline. This is a change that was checked into our internal private fork of the SDK a while ago and was not brought back to the upstream SDK. I think it *should* be brought back in this change (given my reasons above) and I think the only reason it wasn't originally was because a dev likely felt time pressure, but just fyi